test/test_torch.py

```diff
@@ -5671,6 +5671,40 @@ def test_index_add_all_dtypes(self):
                     added = zeros.index_add(0, torch.arange(0, size[0], dtype=idx_dtype, device=device), tensor, alpha=-1)
                     self.assertEqual(added, -tensor)
 
+    def test_index_add_correctness(self):
+        # Check whether index_add can get correct result when
+        # dim is -1, alpha is 1, and dtype of index is torch.long,
+        # i.e., using scatter_add
+        for dtype in all_types_and_complex_and(torch.half, torch.bfloat16):
+            if torch.cuda.is_available():
```

Comment:
add `devices` for the test case and try to compare on conditions such as `alpha = 1.0` and `alpha = 2.0 and result /= 2.0`.